MediaPi.Core/Controllers/DevicesController.cs

```diff
@@ -48,29 +49,60 @@ public class DevicesController(
     // POST: api/devices/register
     [AllowAnonymous]
     [HttpPost("register")]
-    [ProducesResponseType(StatusCodes.Status201Created, Type = typeof(Reference))]
-    public async Task<ActionResult<Reference>> Register()
+    [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(DeviceRegisterResponse))]
```

Comment:
This method is "create"
It shall not be used to update existing entities, there is Update for it